### PR TITLE
Add instructions for setting up a jenkins slave in OpenStack

### LINKED ISSUE #519: Document the steps of setting up an openstack instance as a Testgrid slave
**Description:**
$subject.

**Affected Product Version:**
0.9.0-m14

**OS, DB, other environment details and versions:**    
N/A

**Steps to reproduce:**
N/A

### PR DESCRIPTION
**Pull request by azinneera**

### CODE REVIEW

File: jenkins/slave-setup/setup-slave.sh [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+sudo apt-add-repository ppa:ansible/ansible
+sudo apt-get update
+sudo apt-get install ansible
+
+echo "Installing jinja2"
+sudo apt-get install python-pip
+sudo pip install https://pypi.python.org/packages/2.7/J/Jinja2/Jinja2-2.8-py2.py3-none-any.whl
+
+echo "Installing Ruby"
+gpg --keyserver hkp://keys.gnupg.net --recv-keys 409B6B1796C275462A1703113804BB82D39DC0E3 7D2BAF1CF37B13E2069D6956105BD0E739499BDB
```
Comment (pasindujw) [inline jenkins/slave-setup/setup-slave.sh]: Do we need to add the key to the repo?

File: jenkins/slave-setup/setup-slave.sh [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+#
+# Run this script with changes requested in the README.md
+#
+###########################################################
+
+sudo apt-get update
+sudo apt-get install unzip
+
+echo "Installing terraform"
+wget https://releases.hashicorp.com/terraform/0.11.3/terraform_0.11.3_linux_amd64.zip
```
Comment (kasunbg) [inline jenkins/slave-setup/setup-slave.sh]: retry 3 times with --retries=3

File: jenkins/slave-setup/setup-slave.sh [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+curl -sSL https://get.rvm.io -o rvm.sh
+cat rvm.sh | bash -s stable
+source ~/.rvm/scripts/rvm
+rvm install ruby --default
+
+echo "Installing python net-addr"
+sudo apt-get install python-netaddr
+
+echo "Installing kubectl"
+curl -LO https://storage.googleapis.com/kubernetes-release/release/v1.8.2/bin/linux/amd64/kubectl
```
Comment (kasunbg) [inline jenkins/slave-setup/setup-slave.sh]: Do not mix curl and wget in the same script.. :)

File: jenkins/slave-setup/setup-slave.sh [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+chmod +x ./kubectl
+sudo mv ./kubectl /usr/local/bin/kubectl
+
+echo "Installing git"
+sudo apt-get install git
+
+echo "Installing Java"
+sudo add-apt-repository ppa:webupd8team/java
+sudo apt-get update
+sudo apt-get install oracle-java8-installer
```
Comment (kasunbg) [inline jenkins/slave-setup/setup-slave.sh]: Do you know the version of the JDK this downloads?

File: jenkins/slave-setup/setup-slave.sh [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+kubectl version
+java -version
+
+echo "Generate SSH keys"
+ssh-keygen
+cat ~/.ssh/id_rsa.pub
+
+#Prepare for jenkins slave
+echo "Adding testgrid-live public certificate to slave keystore"
+openssl s_client -connect <server-ip-or-hostname>:443 < /dev/null | sed -ne '/-BEGIN CERTIFICATE-/,/-END CERTIFICATE-/p' > server.crt
```
Comment (kasunbg) [inline jenkins/slave-setup/setup-slave.sh]: Replace <server-ip-or-hostname> with correct hostname.

File: jenkins/slave-setup/setup-slave.sh [hunk 6]
```diff
@@ -0,0 +1,72 @@
...
+$JAVA_HOME/bin/keytool -import -alias testgrid-live.private.wso2.com -keystore $JAVA_HOME/jre/lib/security/cacerts -file server.crt
+
+sudo mkdir /testgrid
+sudo chown ubuntu:ubuntu /testgrid
+mkdir /testgrid/jenkins-home
+wget https://<jenkins-master-url>/jnlpJars/slave.jar --no-check-certificate
```
Comment (kasunbg) [inline jenkins/slave-setup/setup-slave.sh]: <jenkins-master-url> is missing. Please add a global variable at the top, and refer it here.

File: jenkins/slave-setup/setup-slave.sh (hunk 1 above)
Comment (azinneera) [reply jenkins/slave-setup/setup-slave.sh]: This is publicly available and hence IMHO there is no harm pushing this

File: jenkins/slave-setup/setup-slave.sh (hunk 4 above)
Comment (azinneera) [reply jenkins/slave-setup/setup-slave.sh]: This downloads the latest version as of now. The current version being downloaded is jdk 1.8.0_161

**Pull request merged**